Check build fails with flags `--no-default-features --features api`

While working on #7200 I ran `cargo hack check --feature-powerset --clean-per-run` and found that the build string in the issue title causes a build failure. 

```
error[E0432]: unresolved import `crate::sources::host_metrics`
 --> src/api/schema/metrics/host.rs:3:21
  |
3 | use crate::sources::host_metrics::HostMetricsConfig;
  |                     ^^^^^^^^^^^^ could not find `host_metrics` in `sources`

error[E0432]: unresolved import `uuid`
  --> src/api/tap.rs:16:5
   |
16 | use uuid::Uuid;
   |     ^^^^ use of undeclared crate or module `uuid`

error[E0433]: failed to resolve: could not find `host_metrics` in `sources`
   --> src/api/schema/metrics/host.rs:265:23
    |
265 |         Self(sources::host_metrics::HostMetricsConfig::default())
    |                       ^^^^^^^^^^^^ could not find `host_metrics` in `sources`

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `vector`

To learn more, run the command again with --verbose.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -345,6 +345,7 @@ api = [
   "base64",
   "itertools",
   "warp",
+  "uuid",
 ]
 
 # API client

--- a/src/api/schema/metrics/mod.rs
+++ b/src/api/schema/metrics/mod.rs
@@ -2,13 +2,15 @@ mod errors;
 mod events_in;
 mod events_out;
 pub mod filter;
-mod host;
 mod processed_bytes;
 mod processed_events;
 mod sink;
 pub mod source;
 mod transform;
 mod uptime;
+
+#[cfg(feature = "sources-host_metrics")]
+mod host;
 
 use async_graphql::{validators::IntRange, Interface, Object, Subscription};
 use chrono::{DateTime, Utc};
@@ -18,7 +20,6 @@ pub use errors::{ComponentErrorsTotal, ErrorsTotal};
 pub use events_in::{ComponentEventsInThroughput, ComponentEventsInTotal, EventsInTotal};
 pub use events_out::{ComponentEventsOutThroughput, ComponentEventsOutTotal, EventsOutTotal};
 pub use filter::*;
-pub use host::HostMetrics;
 pub use processed_bytes::{
     ComponentProcessedBytesThroughput, ComponentProcessedBytesTotal, ProcessedBytesTotal,
 };
@@ -29,6 +30,9 @@ pub use sink::{IntoSinkMetrics, SinkMetrics};
 pub use source::{IntoSourceMetrics, SourceMetrics};
 pub use transform::{IntoTransformMetrics, TransformMetrics};
 pub use uptime::Uptime;
+
+#[cfg(feature = "sources-host_metrics")]
+pub use host::HostMetrics;
 
 #[derive(Interface)]
 #[graphql(field(name = "timestamp", type = "Option<DateTime<Utc>>"))]
@@ -43,6 +47,7 @@ pub struct MetricsQuery;
 
 #[Object]
 impl MetricsQuery {
+    #[cfg(feature = "sources-host_metrics")]
     /// Vector host metrics
     async fn host_metrics(&self) -> HostMetrics {
         HostMetrics::new()

--- a/src/api/schema/metrics/mod.rs
+++ b/src/api/schema/metrics/mod.rs
@@ -31,9 +31,6 @@ pub use source::{IntoSourceMetrics, SourceMetrics};
 pub use transform::{IntoTransformMetrics, TransformMetrics};
 pub use uptime::Uptime;
 
-#[cfg(feature = "sources-host_metrics")]
-pub use host::HostMetrics;
-
 #[derive(Interface)]
 #[graphql(field(name = "timestamp", type = "Option<DateTime<Utc>>"))]
 pub enum MetricType {
@@ -49,8 +46,8 @@ pub struct MetricsQuery;
 impl MetricsQuery {
     #[cfg(feature = "sources-host_metrics")]
     /// Vector host metrics
-    async fn host_metrics(&self) -> HostMetrics {
-        HostMetrics::new()
+    async fn host_metrics(&self) -> host::HostMetrics {
+        host::HostMetrics::new()
     }
 }
 

